### PR TITLE
blob encode/decode API, tests, plus placeholder implementation

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Establishes the interface for blob encoding & decoding, and introduces a working placeholder implementation which will be replaced by a followup PR implementing the v2 method from the spec: https://github.com/ethereum-optimism/optimism/pull/8745

**Tests**

Unit tests to establish blob encoding / decoding work as expected under a variety of conditions and invalid data is appropriately detected.

### CODE REVIEW

File: op-service/eth/blob.go [hunk 1]
```diff
@@ -66,3 +67,67 @@ func KZGToVersionedHash(commitment kzg4844.Commitment) (out common.Hash) {
 func VerifyBlobProof(blob *Blob, commitment kzg4844.Commitment, proof kzg4844.Proof) error {
 	return kzg4844.VerifyBlobProof(*blob.KZGBlob(), commitment, proof)
 }
+
+// FromData encodes the given input data into this blob. The encoding scheme is as follows:
+//
+// First, field elements are encoded as big-endian uint256 in BLS modulus range. To avoid modulus
+// overflow, we can't use the full 32 bytes, so we write data only to the topmost 31 bytes of each.
+// TODO: we can optimize this to get a bit more data from the blobs by using the top byte
```
Comment (tynes) [inline op-service/eth/blob.go:75]: Did you mean to leave this todo here?

**Comment by tynes** [comment]
I am confused if this implements the spec or not plus the leftover todos. Is this ready for review as is or do we need to update the spec?

**Comment by roberto-bayardo** [comment]
> I am confused if this implements the spec or not plus the leftover todos. Is this ready for review as is or do we need to update the spec?

No it does not implement the latest spec, it's a placeholder (prior) implementation until the latest one goes in so that everything will compile / build and can be tested.

The ref implementation (https://github.com/roberto-bayardo/optimism/pull/7)  is building on this PR.

File: op-service/eth/blob.go (hunk 1 above)
Comment (roberto-bayardo) [reply op-service/eth/blob.go:75]: Yeah see other reply re: this code is a placeholder for the v2 spec implementation building on this PR.

**Comment by tynes** [comment]
> > I am confused if this implements the spec or not plus the leftover todos. Is this ready for review as is or do we need to update the spec?
> 
> No it does not implement the latest spec, it's a placeholder (prior) implementation until the latest one goes in so that everything will compile / build and can be tested.
> 
> The ref implementation ([roberto-bayardo#7](https://github.com/roberto-bayardo/optimism/pull/7)) is building on this PR.

How would you like to proceed here? Complete the implementation there and then pull it in here?

**Comment by roberto-bayardo** [comment]
> How would you like to proceed here? Complete the implementation there and then pull it in here?

Merging this one now would make it possible to proceed with reviewing some of the other PRs that are blocked by having this in place -- not sure if tentative code like this is ok to merge though. Either way the other PR is probably pretty close to ready (cc @angel-ding-cb)

**Review by tynes — APPROVED** [review]
I am ok with merging this as a placeholder for the v2 to be implemented on top

**Pull request merged**